addons/web/controllers/webclient.py

```diff
@@ -70,19 +71,30 @@ def translations(self, unique, mods=None, lang=None):
         if lang and lang not in {code for code, _ in request.env['res.lang'].sudo().get_installed()}:
             lang = None
 
-        translations_per_module, lang_params = request.env["ir.http"].get_translations_for_webclient(mods, lang)
+        current_hash = request.env["ir.http"].with_context(cache_translation_data=True).get_web_translations_hash(mods, lang)
 
         body = {
             'lang': lang,
-            'lang_parameters': lang_params,
-            'modules': translations_per_module,
-            'multi_lang': len(request.env['res.lang'].sudo().get_installed()) > 1,
+            'hash': current_hash,
         }
+        if current_hash == hash:
+            body['no_change'] = True
```

Comment:
I don't think it's necessary. The caller can compare the received hash with the one he sent.

Reply:
it would make the shape of the returned object simpler